typescript/packages/plugins/mode-governance/src/mode-governance.service.ts

```diff
@@ -0,0 +1,110 @@
+import { EVMWalletClient } from "@goat-sdk/wallet-evm";
+import { parseUnits, formatUnits } from "viem";
+import { VOTING_ESCROW_ABI } from "./abi";
+import {
+    MODE_VOTING_ESCROW,
+    BPT_VOTING_ESCROW,
+} from "./constants";
+import {
+    StakeParameters,
+    GetStakeInfoParameters,
+    GetBalanceParameters,
+} from "./parameters";
+
+export class ModeGovernanceService {
+    @Tool({
+        description: "Stake MODE or BPT tokens in the Mode governance system. Requires MODE or BPT tokens to be approved first.",
+    })
+    async stakeTokens(walletClient: EVMWalletClient, parameters: StakeParameters) {
+        const escrowAddress = parameters.tokenType === "MODE" ? MODE_VOTING_ESCROW : BPT_VOTING_ESCROW;
+        
+        const stakeHash = await walletClient.sendTransaction({
+            to: escrowAddress,
+            abi: VOTING_ESCROW_ABI,
+            functionName: "createLock",
+            args: [parseUnits(parameters.amount, 18)],
+        });
+
+        return stakeHash.hash;
+    }
+
+    @Tool({
+        description: "Get staking information including lock period and voting power",
+    })
+    async getStakeInfo(walletClient: EVMWalletClient, parameters: GetStakeInfoParameters) {
+        const escrowAddress = parameters.tokenType === "MODE" ? MODE_VOTING_ESCROW : BPT_VOTING_ESCROW;
+        const userAddress = await walletClient.getAddress();
+
+        const tokenIds = (await walletClient.read({
+            address: escrowAddress,
+            abi: VOTING_ESCROW_ABI,
+            functionName: "ownedTokens",
+            args: [userAddress],
+        })) as unknown as bigint[];
+
+        if (tokenIds.length === 0) {
+            return {
+                stakedAmount: "0",
+                startTime: 0,
+                votingPower: "0",
+                isVoting: false
+            };
+        }
+
+        const tokenId = tokenIds[0];
+        const [amount, start] = (await walletClient.read({
+            address: escrowAddress,
+            abi: VOTING_ESCROW_ABI,
+            functionName: "locked",
+            args: [tokenId],
+        })) as unknown as [bigint, bigint];
+
+        const votingPower = await walletClient.read({
+            address: escrowAddress,
+            abi: VOTING_ESCROW_ABI,
+            functionName: "votingPower",
+            args: [tokenId],
+        }) as unknown as bigint;
+
+        const isVoting = await walletClient.read({
+            address: escrowAddress,
+            abi: VOTING_ESCROW_ABI,
+            functionName: "isVoting",
+            args: [tokenId],
+        }) as unknown as boolean;
+
+        return {
+            stakedAmount: formatUnits(amount, 18),
+            startTime: Number(start),
+            votingPower: formatUnits(votingPower, 18),
+            isVoting
+        };
+    }
+
+    @Tool({
+        description: "Get the voting power for any address",
+    })
+    async getBalance(walletClient: EVMWalletClient, parameters: GetBalanceParameters) {
```

Comment:
I would update the name of the tool. `getBalance` is too generic.

Maybe something like `get_mode_governance_voting_power`

Reply:
true , sounds good